compiler/sem/sempass2.nim

```diff
@@ -1313,8 +1313,11 @@ proc track(tracked: PEffects, n: PNode) =
       if iterCall[1].typ != nil and
          iterCall[1].typ.skipTypes(abstractVar).kind notin {tyVarargs, tyOpenArray}:
         createTypeBoundOps(tracked, iterCall[1].typ, iterCall[1].info)
-    
+
     if tracked.owner.kind != skMacro and iterCall.kind in nkCallKinds and
+       iterCall[0].typ != nil and # XXX: untyped nkForStmt can reach here
```

Comment:
Maybe adding a known issue test for the bug surfaced by `typetraits.elementType` will help us get to the bottom of things. As we suspect this issue will show up again in different ways.

Reply:
Done :)